src/compiler/checker.ts

```diff
@@ -700,6 +700,7 @@ namespace ts {
         const anyType = createIntrinsicType(TypeFlags.Any, "any");
         const autoType = createIntrinsicType(TypeFlags.Any, "any");
         const wildcardType = createIntrinsicType(TypeFlags.Any, "any");
+        const erasedType = createIntrinsicType(TypeFlags.Any | TypeFlags.Never | TypeFlags.Unknown, "any");

```

Comment:
Wow, a schizophrenic type. We have an implicit assumption everywhere that only one of these flags will be set--with a few exceptions such as `TypeFlags.EnumLiteral`. Are you confident there are no places we'll get confused by all three being set in this type? Tests for these now become order dependent.

Reply:
Yep, I'm literally abusing the order dependence of handling each of these to avoid special-casing the handling of this type in most type construction places (partially because we're essentially out of normal type flags). Generally speaking we handle Any, then Never, then Unknown (if we handle both), so it's OK, since this marker is mostly any-like (except in unions/intersections, where we handle unknown or never, and _then_ any, which is what causes the desirable vaporization in both structures). At the start, I _also_ wanted to add `ObjectFlags.NonInferrableType` to it, but that has the side effect of cases like `<T extends T[]>() => T` where we currently infer `any[]` becoming `unknown`. I don't think `any[]` is _great_ (it injects `any` into the program without the user ever explicitly writing it, and it's not an `noImplicitAny` error!), but `unknown` isn't really much better. I think the ideal would be `unknown[]` - but that relies on knowing that `T[]` is covariant on `T`... I guess since we use `getVariances` in inference already, that should be safe. So In theory I could get what I want by mapping this erased type to `unknown` in covariant positions and `never` in contravariant positions, when it persists into the constructed type (rather than just vaporizing in a union or intersection and letting it persist beyond that as a psuedo-`any`). I could also make this a special type parameter, and handle all this with a special instantiator that tracks some context of the instantiation - in fact, to do the variance bit, I probably _need_ to.